### PR TITLE
ISPN-8829 LockedStream doesn't override BaseCacheStream methods

### PR DESCRIPTION
**Pull request by wburns**
* Added methods to interface

https://issues.jboss.org/browse/ISPN-8829

### CODE REVIEW

File: core/src/main/java/org/infinispan/LockedStream.java [hunk 1]
```diff
@@ -135,6 +136,42 @@ default void forEach(SerializableBiConsumer<Cache<K, V>, ? super CacheEntry<K, V
       return invokeAll((BiFunction<Cache<K, V>, ? super CacheEntry<K, V>, R>) biFunction);
    }
 
+   /**
+    * {@inheritDoc}
+    */
+   @Override
+   LockedStream sequentialDistribution();
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/LockedStream.java]: All return types need to be in the generic form LockedStream<K, V> to preserve type information in chained invocations

Comment (wburns) [reply core/src/main/java/org/infinispan/LockedStream.java]: Oops I didn't mean to forget those 👍

**Comment by wburns** [comment]
Fixed and also updated the old methods that had it incorrect as well.

**Comment by tristantarrant** [comment]
Looks good, waiting for CI to complete

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**